Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1

```diff
@@ -55,50 +56,86 @@ $credential = New-Object 'System.Net.NetworkCredential' -ArgumentList $machineUs
 
         Output-ResponseLogs -operationName "deployment" -fqdn $fqdn -deploymentResponse $deploymentResponse
 
         $response = $deploymentResponse
-    }
+        $status = $response.Status
+        Write-Output "Deployment Status for machine $fdqn : $status"
 
-    $logs = New-Object 'System.Collections.Generic.List[System.Object]'         
-    $resourceOperationLog = New-OperationLog -Content $log
-    $logs.Add($resourceOperationLog)
+        $logs = New-Object 'System.Collections.Generic.List[System.Object]'         
+        $resourceOperationLog = New-OperationLog -Content $log
+        $logs.Add($resourceOperationLog)
 
-    Complete-ResourceOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -ResourceOperationId $resOperationId -Status $response.Status -ErrorMessage $response.Error -Logs $logs -ErrorAction Stop -Connection $connection
+        Complete-ResourceOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -ResourceOperationId $resOperationId -Status $response.Status -ErrorMessage $response.Error -Logs $logs -ErrorAction Stop -Connection $connection
     
-    if ($response.Status -ne "Passed")
+        if ($response.Status -ne "Passed")
+        {
+            Complete-EnvironmentOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -Status "Failed" -ErrorAction Stop -Connection $connection
+
+            throw $response.Error;
+        }
+    }
+}
+else
+{
+    $Jobs = New-Object "System.Collections.Generic.Dictionary``2[int, string]"
+
+    foreach($resource in $resources)
     {
-        Complete-EnvironmentOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -Status "Failed" -ErrorAction Stop -Connection $connection
+        $machine = $resource.Name
+	
+        Write-Output "Deployment will be Started for - $machine"
 
-        throw $response.Error;
+        $job = Start-Job -ScriptBlock $RunPowershellJob -ArgumentList $environmentName, $envOperationId, $machine, $scriptPath, $port, $scriptArguments, $initializationScriptPath, $credential, $connection
+
+        $Jobs.Add($job.Id, $machine)
+    
+        Write-Output "Deployment Started for  - $machine"
+    }
+    While (Get-Job)
+    {
+         Start-Sleep 10 
+         foreach($job in Get-Job)
+         {
+             if($job.State -ne "Running")
+             {
+                 $output = Receive-Job -Job $job
+                 $result = $output.DeploymentLog
+                 $status = $output.Status
+
+                 if($status -ne "Passed")
+                 {
+                     $envOperationStatus = "Failed"
+                 }
+            
+                 $machineName = $Jobs.Item($job.Id)
+
+                 Output-ResponseLogs -operationName "Deployment" -fqdn $machineName -deploymentResponse $output
+                 Write-Output "Result for machine $machineName : $result"
```

Comment:
We should be removing these write-output's which are just printing duplicate information which are already in the write-verbose in Output-ResponseLogs. In my opinion it is cluttering our logs. We should not print this on console, as it is not live and does not make sense. Just printing "Completed" will suffice. Any thoughts?